src/main/java/org/candlepin/model/Consumer.java

```diff
@@ -1024,4 +988,32 @@ public Consumer setServiceType(String serviceType) {
         this.serviceType = serviceType;
         return this;
     }
+
+    public Map<String, String> getEnvironmentIds() {
+        return this.environmentIds;
+    }
+
+    public Consumer setEnvironmentIds(Map<String, String> environmentIds) {
+        this.environmentIds = environmentIds;
```

Comment:
Let's use the same logic of adding map elements as other places (e.g. Product.setAttributes), by clearing the original map, and then re-populating it with the elements of the incoming map. Assigning the incoming map directly can be dangerous because someone that holds it's reference would be able to update it later on, which would be reflected on the map in the Consumer object

Reply:
Agreed. Done.